addons/account/models/account_account.py

```diff
@@ -465,13 +576,48 @@ def _onchange_user_type_id(self):
         elif self.internal_group == 'expense' and not self.tax_ids:
             self.tax_ids = self.company_id.account_purchase_tax_id
 
+    @api.onchange('code')
+    def _onchange_code(self):
+        code, name = self._split_code_name(self.code)
+        if name:
+            self.name = name
+            self.code = code
+
+    @api.onchange('name')
+    def _onchange_name(self):
+        code, name = self._split_code_name(self.name)
+        if code:
+            self.name = name
+            self.code = code
+
     def name_get(self):
         result = []
         for account in self:
-            name = account.code + ' ' + account.name
+            name = account.code + ' ' + (account.name or '')
             result.append((account.id, name))
         return result
 
+    @api.model
+    def name_create(self, name):
+        # name_create is possible when importing from a related model
+        if 'import_file' in self.env.context:
+            code, name = self._split_code_name(name)
+            return self.create({'code': code, 'name': name}).name_get()[0]
+        # Enforce no_quick_create from forms to ensure that the account.account form view pops up
+        # even when no_quick_create is mistakenly forgotten from the view
+        raise ValueError(_("The values for the created account need to be verified."))
+
+    def _split_code_name(self, code_name):
+        results = ACCOUNT_REGEX.match(code_name or '')
+        return results[1], results[2].strip()
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
I would remove this feature, it's not necessary; it's weird to import code and name in the same field.